common/persistence/sql/sql_task_store.go

```diff
@@ -252,8 +252,8 @@ func (m *sqlTaskStore) UpdateTaskList(
 		LastUpdated:             time.Now(),
 		AdaptivePartitionConfig: toSerializationTaskListPartitionConfig(request.TaskListInfo.AdaptivePartitionConfig),
 	}
-	if request.TaskListInfo.Kind == persistence.TaskListKindSticky {
-		tlInfo.ExpiryTimestamp = stickyTaskListExpiry()
+	if persistence.TaskListKindHasTTL(request.TaskListInfo.Kind) {
+		tlInfo.ExpiryTimestamp = time.Now().Add(taskListTTL)
```

Comment:
I'm pretty sure it's wrong that we're using `time.Now()` everywhere in sqlTaskStore instead of the request timestamp, but I'm going to leave that for another PR.